rwengine/src/objects/PickupObject.cpp

```diff
@@ -143,6 +131,43 @@ void PickupObject::tick(float dt) {
         }
     }
 
+    int coronaColourId = 0;
+#if 0
+    if (modelinfo->name == "colt45")
+        coronaColourId = 1;
+    else if (modelinfo->name == "uzi")
+        coronaColourId = 2;
+    else if (modelinfo->name == "shotgun")
+        coronaColourId = 3;
+    else if (modelinfo->name == "ak47")
+        coronaColourId = 4;
+    else if (modelinfo->name == "m16")
+        coronaColourId = 5;
+    else if (modelinfo->name == "sniper")
+        coronaColourId = 6;
+    else if (modelinfo->name == "rocket")
+        coronaColourId = 7;
+    else if (modelinfo->name == "flame")
+        coronaColourId = 8;
+    else if (modelinfo->name == "molotov")
+        coronaColourId = 9;
+    else if (modelinfo->name == "grenade")
+        coronaColourId = 10;
+    else if (modelinfo->name == "bodyarmour" || modelinfo->name == "bribe")
+        coronaColourId = 12;
+    else if (modelinfo->name == "info" || modelinfo->name == "killfrenzy")
+        coronaColourId = 13;
+    else if (modelinfo->name == "health" || modelinfo->name == "bonus")
+        coronaColourId = 14;
+#endif
+    float timeInMs = engine->getGameTime();
+    float m = 0.5f * (sin(timeInMs * 3.0664064f) * 0.3f + 0.3f);
+    float *colour = glm::value_ptr(colours[coronaColourId]);
+    m_corona->particle.colour = glm::vec4(colour[0] * m,
+                                          colour[1] * m,
+                                          colour[2] * m,
+                                          colour[3] * m);
```

Comment:
vec4 has an overload on the * operator for scalar multiplication, you can just write `m_corona->particle.colour = colours[coronaColourId] * m` instead of doing it component-wise.